design/ceph-mon-pv.md

```diff
@@ -0,0 +1,139 @@
+## Non-PV monitor storage
+
+Not all monitor storage requirements should be satisfied using PV storage. Most
+notably log and crash data. These data may continue to be stored in HostPath
+volume mounts. This will continue to be based on `dataDirHostPath`.
+
+## Host networking
+
+Currently Ceph monitor pods may be deployed using host-based networking, or
+placed behind a service using a portable network address. This poses a challenge
+when migrating a monitor (and it's PV) to a new node which has been deployed
+using host-based networking as the monitor's address will no longer valid.
+
+Support for migration of monitors with host-based networking is future work. To
+simplify the initial implementation, host-based networking will not be supported
+for non-local PVs.
+
+# StatefulSet pod management
+
+Monitor pods will be managed by a StatefulSet rather than Deployments which are
+currently use. This change offers (1) simplified naming for DNS and pods (e.g.
+`ceph-mon-a` rather than `ceph-mon-a-3ab423fb`), and (2) StatefulSets are able
+to handle much of the complexity of managing PVs including the mechanics of
+PV/pod association and scheduling.
+
+StatefulSets will be used to manage monitors pods in a 1-1 mapping between a pod
+and StatefulSet. This choice is made to retain maximum flexibility on naming,
+placement, and life cycle management of the pods by Rook.
+
+## Scheduling
+
+Rook currently makes explicit scheduling decisions for monitors. Standard node
+selector is used to control placement for non-local PVs (or dynamically
+provisions local PVs [0]). For statically provisioned local PVs, the Rook
+monitor scheduler will likely need to be made aware of available PVs.
+
+If we want to continue to support the current behavior of Rook behavior
+effectively like a HostPath dynamic provisioner then we could continue to use
+the HostPath mechanism, but we could also allow Rook to create the local PV
+itself. Effectively like this:
+
+```yaml
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: ceph-mon-a-pv
+  labels:
+    vol: ceph-mon-a-pv
+spec:
+  capacity:
+    storage: 1Gi
+  volumeMode: Filesystem
+  accessModes:
+    - ReadWriteOnce
+  storageClassName: local-storage
+  local:
+    path: /opt/mon-a-0-0
+  nodeAffinity:
+    required:
+      nodeSelectorTerms:
+      - matchExpressions:
+        - key: kubernetes.io/hostname
+          operator: In
+          values:
+            - worker1
+---
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: local-storage
+provisioner: kubernetes.io/no-provisioner
+volumeBindingMode: WaitForFirstConsumer
+```
+
+# Upgrades
+
+Because of the switch from Deployments to Stateful sets, the most
+straightforward method for upgrading the cluster will be to perform a
+controlled replacement of each existing monitor deployment.
+
+There will be other upgrade scenarios that need to be addressed but as long
+as networking and paths between an existing Deployment and the replacement
```

Comment:
Just like you mentioned above, I think the hostNetwork case is probably the trickiest.